test/failed-directories.test.js

```diff
@@ -319,14 +387,146 @@ describe('cosmiconfig', () => {
 
         expect.assertions(2);
         try {
-          search(true);
+          search({ sync: true });
         } catch (err) {
           expect(err.name).toBe('SyntaxError');
         }
-        return search(false).catch(err => {
+        return search({ sync: false }).catch(err => {
           expect(err.name).toBe('SyntaxError');
         });
       });
+
+      it('returns an undefined config, file path, and isEmpty = true for an empty json', () => {
+        function readFile(searchPath) {
+          switch (searchPath) {
+            case absolutePath('a/b/c/d/e/f/package.json'):
+            case absolutePath('a/b/c/d/e/f/.foorc'):
+            case absolutePath('a/b/c/d/e/f/.foorc.yaml'):
+            case absolutePath('a/b/c/d/e/f/.foorc.yml'):
+            case absolutePath('a/b/c/d/e/f/.foorc.js'):
+              throw { code: 'ENOENT' };
+            case absolutePath('a/b/c/d/e/f/.foorc.json'):
+              return '';
+            default:
+              throw new Error(`irrelevant path ${searchPath}`);
+          }
+        }
+        mockReadFile(true, readFile);
+        mockReadFile(false, readFile);
+
+        expect.assertions(2);
+
+        const expectedResult = {
+          config: undefined,
+          filepath: absolutePath('a/b/c/d/e/f/.foorc.json'),
+          isEmpty: true,
+        };
+        expect(search({ ignoreEmpty: false, sync: true })).toEqual(
+          expectedResult
+        );
+        return expect(
+          search({ ignoreEmpty: false, sync: false })
+        ).resolves.toEqual(expectedResult);
+      });
+
+      it('returns an undefined config, file path, and isEmpty = true for empty yaml', () => {
+        function readFile(searchPath) {
+          switch (searchPath) {
+            case absolutePath('a/b/c/d/e/f/package.json'):
+            case absolutePath('a/b/c/d/e/f/.foorc'):
+            case absolutePath('a/b/c/d/e/f/.foorc.json'):
+            case absolutePath('a/b/c/d/e/f/.foorc.yml'):
+            case absolutePath('a/b/c/d/e/f/.foorc.js'):
+              throw { code: 'ENOENT' };
+            case absolutePath('a/b/c/d/e/f/.foorc.yaml'):
+              return '';
+            default:
+              throw new Error(`irrelevant path ${searchPath}`);
+          }
+        }
+        mockReadFile(true, readFile);
+        mockReadFile(false, readFile);
+
+        expect.assertions(2);
+
+        const expectedResult = {
+          config: undefined,
+          filepath: absolutePath('a/b/c/d/e/f/.foorc.yaml'),
+          isEmpty: true,
+        };
+        expect(search({ ignoreEmpty: false, sync: true })).toEqual(
+          expectedResult
+        );
+        return expect(
+          search({ ignoreEmpty: false, sync: false })
+        ).resolves.toEqual(expectedResult);
+      });
+
+      it('returns an undefined config, file path, and isEmpty = true for empty yml', () => {
```

Comment:
I don't think this is required. Testing for `yaml` should do. Otherwise maybe we should look into integrating [`jest-in-case`](https://github.com/atlassian/jest-in-case).

Reply:
I think prior to shortening that portion of the code, it was necessary for 100% coverage.  You're right it can be removed now, but so can the empty `js` and `yml`.  Should we remove those as well?

Reply:
No, let's keep the other ones. Might be useful if we ever refactor that section again.